src/ILCompiler/src/RdXmlRootProvider.cs

```diff
@@ -123,34 +124,63 @@ private void ProcessMethodDirective(IRootingServiceProvider rootProvider, Module
+                throw new Exception("The \"Name\" attribute is required on the \"Method\" Runtime Directive.");
 
+            var parameter = new List<TypeDesc>();
             var instArgs = new List<TypeDesc>();
             foreach (var element in methodElement.Elements())
             {
                 switch (element.Name.LocalName)
                 {
+                    case "Parameter":
+                        string paramName = element.Attribute("Name").Value;
+                        parameter.Add(containingModule.GetTypeByCustomAttributeTypeName(paramName));
+                        break;
                     case "GenericArgument":
                         string instArgName = element.Attribute("Name").Value;
                         instArgs.Add(containingModule.GetTypeByCustomAttributeTypeName(instArgName));
                         break;
                     default:
-                        throw new NotSupportedException();
+                        throw new NotSupportedException($"\"{element.Name.LocalName}\" is not a supported Runtime Directive.");
                 }
             }
 
-            if (instArgs.Count != method.Instantiation.Length)
-                throw new Exception();
-
-            if (instArgs.Count > 0)
+            static bool SignatureMatches(MethodDesc method, List<TypeDesc> parameter)
             {
-                var methodInst = new Instantiation(instArgs.ToArray());
-                method = method.MakeInstantiatedMethod(methodInst);
+                if (parameter.Count != method.Signature.Length)
+                    return false;
+
+                for (int i = 0; i < method.Signature.Length; i++)
+                {
+                    if (method.Signature[i] != parameter[i])
+                    {
+                        return false;
+                    }
+                }
+                return true;
             }
 
-            RootingHelpers.TryRootMethod(rootProvider, method, "RD.XML root");
+            string methodName = methodNameAttribute.Value;
+            foreach (var m in containingType.GetMethods())
+            {
+                var method = m;
+                if (method.Name != methodName)
+                    continue;
+
+                if (parameter.Count > 0 && !SignatureMatches(method, parameter))
+                    continue;
+
+                if (instArgs.Count != method.Instantiation.Length)
+                    throw new Exception($"Could not instantiate Method {method} specified by a Runtime Directive. Method takes {method.Instantiation.Length} generic argument(s) but {instArgs.Count} were provided.");
+
+                if (instArgs.Count > 0)
+                {
+                    var methodInst = new Instantiation(instArgs.ToArray());
+                    method = method.MakeInstantiatedMethod(methodInst);
+                }
+
+                RootingHelpers.TryRootMethod(rootProvider, method, "RD.XML root");
+            }
```

Comment:
I think the check for whether anything was rooted here would be useful. It makes it easier to troubleshoot if one messed up a signature or has a typo in the method name.